File: 2016-maximum-difference-between-increasing-elements/2016-maximum-difference-between-increasing-elements.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maximumDifference = function(nums) {
    let min = nums[0], res = -1
    for (let i=1; i < nums.length; i++) {
        res = Math.max(res, nums[i] - min)
        min = Math.min(min, nums[i])
    }
    return res === 0 ? -1 : res
};

File: 2016-maximum-difference-between-increasing-elements/2016-maximum-difference-between-increasing-elements.py
class Solution:
    def maximumDifference(self, nums: List[int]) -> int:
        mn, res = nums[0], -1
        for i in range(1, len(nums)):
            mn = min(mn, nums[i])
            res = max(res, nums[i] - mn)
        return res if res != 0 else -1